### DISCUSSION TITLE
When user buy a subscription and no payment method added then amount deduct but receipt not received.

### DISCUSSION THREAD

**Discussion by deepakmemor** [hyochan/react-native-iap#1451] [Q&A]
When user click on buy subscription and purchase initiate then library ask to add purchase method and redirect to add payment form, after adding card detail payment deduct but its not redirect to app and receipt not received.

**Comment by hyochan** [comment]
Are you saying `purchaseUpdatedListener` is not firing? You might experience this when you rely on `promises`. This was the [main change in react-native-iap v3](https://medium.com/dooboolab/react-native-iap-v3-1259e0b0c017).

**Reply by deepakmemor** [reply]
Here is my code.

 purchaseUpdateSubscription = purchaseUpdatedListener(
      async (purchase: InAppPurchase | SubscriptionPurchase) => {
       
        if (purchase && purchase.originalTransactionIdentifierIOS) {
          const receipt = purchase.transactionReceipt;
     } 
      },
    );
    purchaseErrorSubscription = purchaseErrorListener(
      (error: PurchaseError) => {

      },
    );

    return (() => {
      if (purchaseUpdateSubscription) {
        purchaseUpdateSubscription.remove();
        purchaseUpdateSubscription = null;
      }
      if (purchaseErrorSubscription) {
        purchaseErrorSubscription.remove();
        purchaseErrorSubscription = null;
      }
    })



In this case purchaseUpdatedListener is called multiple times but it should be call  run only after purchase is completed
